packages/leancode_lint/test/lints_test_app/lib/bloc_related_class_naming_test.dart

```diff
@@ -0,0 +1,100 @@
+import 'package:lints_test_app/bloc_related_class_naming_utils.dart';
+
+// Invalid state name is flagged
+class Test1Cubit extends Cubit<WrongStateName> {
+  Test1Cubit() : super(WrongStateName());
+}
+
+// expect_lint: bloc_related_class_naming
+final class WrongStateName {}
+
+//////////////////////////////////////////////////////////////////////////
+
+// Invalid bloc event name is flagged
+class Test2Bloc extends Bloc<WrongEventName, Test2State> {
+  Test2Bloc() : super(Test2State());
+}
+
+// expect_lint: bloc_related_class_naming
+final class WrongEventName {}
+
+final class Test2State {}
+
+//////////////////////////////////////////////////////////////////////////
+
+// Invalid presentation event name is flagged
+class Test3Bloc extends Bloc<Test3Event, Test3State>
+    with BlocPresentationMixin<Test3State, WrongPresentationEventName> {
+  Test3Bloc() : super(Test3State());
+}
+
+final class Test3Event {}
+
+final class Test3State {}
+
+// expect_lint: bloc_related_class_naming
+final class WrongPresentationEventName {}
+
+//////////////////////////////////////////////////////////////////////////
+
+// Invalid names of enums are flagged
+class Test4Bloc extends Bloc<WrongEnumEvent, WrongEnumState>
+    with BlocPresentationMixin<WrongEnumState, WrongEnumPresentationEvent> {
+  Test4Bloc() : super(WrongEnumState.one);
+}
+
+// expect_lint: bloc_related_class_naming
+enum WrongEnumEvent { one }
+
+// expect_lint: bloc_related_class_naming
+enum WrongEnumState { one }
+
+// expect_lint: bloc_related_class_naming
+enum WrongEnumPresentationEvent { one }
+
+///////////////////////////////////////////////////////////////////////
+
+// No lint when names are correct
+class Correct1Cubit extends Cubit<Correct1State>
+    with BlocPresentationMixin<Correct1State, Correct1Event> {
+  Correct1Cubit() : super(Correct1State());
+}
+
+final class Correct1State {}
+
+final class Correct1Event {}
+
+//////////////////////////////////////////////////////////////////////////
+
+// No lint when names are correct, with event disambiguation
+class Correct2Bloc extends Bloc<Correct2Event, Correct2State>
+    with BlocPresentationMixin<Correct2State, Correct2PresentationEvent> {
+  Correct2Bloc() : super(Correct2State());
+}
+
+final class Correct2Event {}
+
+final class Correct2State {}
+
+final class Correct2PresentationEvent {}
+
+//////////////////////////////////////////////////////////////////////////
+
+// No lint when class is from the same package, but another file
+class SamePackageBloc extends Bloc<ClassFromSamePackage, ClassFromSamePackage>
+    with BlocPresentationMixin<ClassFromSamePackage, ClassFromSamePackage> {
+  SamePackageBloc() : super(ClassFromSamePackage());
+}
+
+//////////////////////////////////////////////////////////////////////////
+
+// No lint when class is from another package
+class AnotherPackageBloc extends Bloc<int, int>
+    with BlocPresentationMixin<int, int> {
+  AnotherPackageBloc() : super(0);
+}
```

Comment:
Please add a case where we have more than one StateClass eg:
```
class Correct1Cubit extends Cubit<Correct1State>
    with BlocPresentationMixin<Correct1State, Correct1Event> {
  Correct1Cubit() : super(Correct1State());
}

selaed class Correct1State {}

final class Correct1Initial {}
final class Correct1Loading {}
final class Correct1Loaded {}
```

Reply:
Additionally I think: 
```
final class Correct1Initial {}
final class Correct1Loading {}
final class Correct1Loaded {}
```
These cases should be legal.

Reply:
Ok makes sense that it follows leancode standards but I could be misleading for external people because in bloc documentation examples uses this format:
```
selaed class ExampleState {}

final class ExampleInitial {}
final class ExampleLoading {}
final class ExampleLoaded {}
```
Maybe we could add some configuration to allow this?